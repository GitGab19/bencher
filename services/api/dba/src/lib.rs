#[macro_use]
extern crate diesel;

pub mod api;
pub mod model;
pub mod schema;
