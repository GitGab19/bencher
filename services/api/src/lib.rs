#[macro_use]
extern crate diesel;

pub mod config;
pub mod endpoints;
pub mod error;
pub mod model;
pub mod schema;
pub mod util;

pub use error::{ApiError, WordStr};
