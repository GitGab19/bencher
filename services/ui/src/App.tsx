import "./styles/styles.scss";

import { createSignal, lazy, Component, createMemo, For } from "solid-js";
import { Routes, Route, useLocation } from "solid-app-router";
import { Navbar } from "./components/site/navbar/Navbar";
import Footer from "./components/site/pages/Footer";
import { projectSlug } from "./components/console/ConsolePage";
import {
	BENCHER_USER_KEY,
	default_user,
	load_user,
	validate_user,
} from "./components/site/util";
import { createStore } from "solid-js/store";

const AuthRoutes = lazy(() => import("./components/auth/AuthRoutes"));
const LandingPage = lazy(() => import("./components/site/pages/LandingPage"));
const PricingPage = lazy(() => import("./components/site/pages/PricingPage"));
const ConsoleRoutes = lazy(() => import("./components/console/ConsoleRoutes"));
const DocsRoutes = lazy(() => import("./components/docs/DocsRoutes"));
const PerfRoutes = lazy(() => import("./components/perf/PerfRoutes"));
const LegalRoutes = lazy(() => import("./components/legal/LegalRoutes"));
const Repo = lazy(() => import("./components/site/pages/Repo"));
const Demo = lazy(() => import("./components/site/pages/Demo"));

const App: Component = () => {
	const location = useLocation();
	const pathname = createMemo(() => location.pathname);

	const [organization_slug, setOrganizationSlug] = createSignal<null | String>(
		null,
	);
	// The project slug can't be a resource because it isn't 100% tied to the URL
	const [project_slug, setProjectSlug] = createSignal<String>(
		projectSlug(pathname),
	);

	const [user, setUser] = createStore(load_user());

	const handleUser = (user): boolean => {
		if (validate_user(user)) {
			window.localStorage.setItem(BENCHER_USER_KEY, JSON.stringify(user));
			setUser(user);
			return true;
		} else {
			return false;
		}
	};

	const removeUser = () => {
		window.localStorage.clear();
		setUser(default_user());
	};

	return (
		<>
			<Navbar
				user={user}
				organization_slug={organization_slug}
				project_slug={project_slug}
				handleProjectSlug={setProjectSlug}
			/>

			<Routes>
				<Route path="/" element={<LandingPage user={user} />} />

				{/* Auth Routes */}
				<Route path="/auth">
					<AuthRoutes
						user={user}
						handleUser={handleUser}
						removeUser={removeUser}
					/>
				</Route>

				{/* Console Routes */}
				<Route path="/console">
					<ConsoleRoutes
						user={user}
						organization_slug={organization_slug}
						project_slug={project_slug}
						handleOrganizationSlug={setOrganizationSlug}
						handleProjectSlug={setProjectSlug}
					/>
				</Route>

				{/* Docs Routes */}
				<Route path="/docs">
					<DocsRoutes />
				</Route>

				{/* Perf Routes */}
				<Route path="/perf">
					<PerfRoutes user={user} />
				</Route>

				{/* Pricing Routes */}
				<Route path="/pricing" element={<PricingPage user={user} />} />

				{/* Legal Routes */}
				<Route path="/legal">
					<LegalRoutes />
				</Route>

				{/* GitHub repo shortcut */}
				<Route path="/repo" element={<Repo />} />

				{/* Calendly demo shortcut */}
				<Route path="/demo" element={<Demo />} />
			</Routes>

			<For each={[...Array(16).keys()]}>{(_k, _i) => <br />}</For>
			<Footer />
		</>
	);
};

export default App;
