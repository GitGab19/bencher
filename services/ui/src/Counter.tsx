import { createSignal, createEffect } from "solid-js";

const [count, setCount] = createSignal(2);

export function Multiplier(prop) {
  return (
    <div>
      <h1>
        {count()} * {prop.by} = {count() * prop.by}
      </h1>
      <button onClick={() => setCount(count() + 1)}>Counter</button>
    </div>
  );
}
