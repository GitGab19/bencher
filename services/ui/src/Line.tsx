import * as Plot from "@observablehq/plot";

const aapl = [
  {
    Date: "2013-05-13T00:00:00.000Z",
    Open: 64.501427,
    High: 65.414284,
    Low: 64.5,
    Close: 64.96286,
    "Adj Close": 50.961628,
    Volume: 79237200,
  },
  {
    Date: "2013-05-14T00:00:00.000Z",
    Open: 64.835716,
    High: 65.028572,
    Low: 63.164288,
    Close: 63.408573,
    "Adj Close": 49.742329,
    Volume: 111779500,
  },
  {
    Date: "2013-05-15T00:00:00.000Z",
    Open: 62.737144,
    High: 63,
    Low: 60.337143,
    Close: 61.264286,
    "Adj Close": 48.060188,
    Volume: 185403400,
  },
  {
    Date: "2013-05-16T00:00:00.000Z",
    Open: 60.462856,
    High: 62.549999,
    Low: 59.842857,
    Close: 62.082859,
    "Adj Close": 48.702328,
    Volume: 150801000,
  },
  {
    Date: "2013-05-17T00:00:00.000Z",
    Open: 62.721428,
    High: 62.869999,
    Low: 61.572857,
    Close: 61.894287,
    "Adj Close": 48.554409,
    Volume: 106976100,
  },
];
const plot = Plot.plot({
  y: {
    grid: true,
  },
  marks: [Plot.line(aapl, { x: "Date", y: "Close" })],
});

export function LinePlot(prop) {
  return (
    <div>
      {plot}  
    </div>
  );
}