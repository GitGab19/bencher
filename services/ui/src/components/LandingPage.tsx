import { LinePlot } from "./plot/LinePlot";

export const LandingPage = () => {
  return (
    <section class="section">
      <div class="container">
        <div class="columns">
          <div class="column">
            <LinePlot />
          </div>
        </div>
      </div>
    </section>
  );
};
