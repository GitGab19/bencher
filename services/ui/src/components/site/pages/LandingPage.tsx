import { Link, useNavigate } from "solid-app-router";
import { createEffect } from "solid-js";
import { BENCHER_TITLE, pageTitle, validate_jwt } from "../util";

const LandingPage = (props) => {
	const navigate = useNavigate();

	createEffect(() => {
		if (validate_jwt(props.user.token)) {
			navigate("/console");
		}

		pageTitle(BENCHER_TITLE);
	});

	return (
		<section class="section is-medium">
			<div class="container">
				<div class="content has-text-centered">
					<h1 class="title">Catch Performance Regressions in CI</h1>
					<h4 class="subtitle">
						Detect and prevent performance regressions before they make it to
						production
					</h4>
					<div class="columns is-centered">
						<div class="column is-half">
							<button
								class="button is-primary is-large is-responsive is-fullwidth"
								onClick={(e) => {
									e.preventDefault();
									navigate("/auth/signup");
								}}
							>
								Start Now
							</button>
						</div>
					</div>
				</div>
			</div>

			<br />
			<br />
			<br />
			<hr />

			<section class="section">
				<div class="container">
					<div class="columns is-centered">
						<div class="column">
							<h2 class="title">How It Works</h2>
						</div>
					</div>
					<br />
					<br />
					<div class="columns is-centered">
						<div class="column">
							<div class="columns">
								<div class="column">
									<div class="content has-text-centered">
										<span class="icon has-text-primary">
											<i
												class="fas fa-tachometer-alt fa-5x"
												aria-hidden="true"
											/>
										</span>
										<h5 class="title">Run your benchmarks</h5>
									</div>
									<div class="content">
										<p>
											Run your benchmarks locally or in CI using your favorite
											tools. The <code>bencher</code> CLI simply wraps your
											existing benchmarking framework and stores its results.
										</p>
										<br />
									</div>
								</div>
							</div>
						</div>
						<br />
						<div class="column">
							<div class="columns">
								<div class="column">
									<div class="content has-text-centered">
										<span class="icon has-text-primary">
											<i class="fas fa-chart-line fa-5x" aria-hidden="true" />
										</span>
										<h5 class="title">Track your benchmarks</h5>
									</div>
									<div class="content">
										<p>
											Track the results of your benchmarks over time. Monitor,
											query, and graph the results using the Bencher web console
											based on the source branch and testbed.
										</p>
									</div>
									<br />
								</div>
							</div>
						</div>
						<br />
						<div class="column">
							<div class="columns">
								<div class="column">
									<div class="content has-text-centered">
										<span class="icon has-text-primary">
											<i class="fas fa-bell fa-5x" aria-hidden="true" />
										</span>
										<h5 class="title">Catch performance regressions</h5>
									</div>
									<div class="content">
										<p>
											Catch performance regressions in CI. Bencher uses state of
											the art, customizable analytics to detect performance
											regressions before they make it to production.
										</p>
									</div>
								</div>
							</div>
						</div>
					</div>
				</div>
			</section>

			<hr />

			<section class="section">
				<div class="container">
					<div class="columns is-centered">
						<div class="column">
							<h2 class="title">Hosting</h2>
						</div>
					</div>
					<br />
					<br />
					<div class="columns is-centered">
						<div class="column">
							<div class="columns">
								<div class="column">
									<div class="content has-text-centered">
										<h2 class="subtitle">Self-Hosted</h2>
										<br />
										<br />
										<br />
										<span class="icon has-text-primary">
											<i class="fas fa-server fa-10x" aria-hidden="true" />
										</span>
									</div>
									<div class="content">
										<p>
											Run Bencher on-prem or in your own cloud. Bencher can be
											deployed on a standalone server, in a Docker container, or
											as part of a Kubernetes cluster.
										</p>
										<br />
									</div>
									<Link href="/docs/how-to/quick-start">
										<button class="button is-fullwidth">Learn More</button>
									</Link>
								</div>
							</div>
						</div>

						<div class="is-divider-vertical" data-content="OR" />

						<div class="column">
							<div class="columns">
								<div class="column">
									<div class="content has-text-centered">
										<h2 class="subtitle">Bencher Cloud</h2>
										<br />
										<br />
										<br />
										<span class="icon has-text-primary is-disabled">
											<i class="fas fa-cloud fa-10x" aria-hidden="true" />
										</span>
									</div>
									<div class="content">
										<p>
											It's {new Date().getFullYear()}, who wants to manage yet
											another serviceâ€½ Let us take care of that for you. All of
											the same great features with none of the hassle.
										</p>
										<br />
									</div>
									<Link href="/docs/how-to/quick-start">
										<button class="button is-fullwidth">Learn More</button>
									</Link>
								</div>
							</div>
						</div>
					</div>
				</div>
			</section>
		</section>
	);
};

export default LandingPage;
