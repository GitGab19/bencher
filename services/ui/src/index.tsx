/* @refresh reload */
import { render } from "solid-js/web";
import { Router } from "solid-app-router";
import { MDXProvider } from "solid-jsx";

import App from "./App";
import mdx from "./md";

render(
  () => (
    <Router>
      <MDXProvider components={{ ...mdx }}>
        <App />
      </MDXProvider>
    </Router>
  ),
  document.getElementById("root") as HTMLElement
);
