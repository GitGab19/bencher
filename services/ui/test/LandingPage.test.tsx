import { describe, expect, test } from "vitest";
import { render } from "solid-testing-library";
import LandingPage from "../src/components/site/pages/LandingPage";
import { defaultUser } from "../src/App";
import { Router } from "solid-app-router";
import { createSignal } from "solid-js";

// https://github.com/vitest-dev/vitest/tree/main/examples/solid
describe("<LandingPage />", () => {
	test("renders", () => {
		const [user, _setUser] = createSignal(defaultUser());
		const { container, unmount } = render(() => (
			<Router>
				<LandingPage user={user} />
			</Router>
		));
		expect(container.innerHTML).toMatchSnapshot();
		unmount();
	});
});
