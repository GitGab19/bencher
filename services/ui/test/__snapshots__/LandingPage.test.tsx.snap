// Vitest Snapshot v1

exports[`<LandingPage /> > renders 1`] = `"<section class=\\"section is-medium\\"><div class=\\"container\\"><div class=\\"content has-text-centered\\"><h1 class=\\"title\\">Catch Performance Regressions in CI</h1><h4 class=\\"subtitle\\">Detect and prevent performance regressions before they make it to production</h4><div class=\\"columns is-centered\\"><div class=\\"column is-half\\"><button class=\\"button is-primary is-large is-responsive is-fullwidth\\">Start Now</button></div></div></div></div><br><br><br><hr><section class=\\"section\\"><div class=\\"container\\"><div class=\\"columns is-centered\\"><div class=\\"column\\"><h2 class=\\"title\\">How It Works</h2></div></div><br><br><div class=\\"columns is-centered\\"><div class=\\"column\\"><div class=\\"columns\\"><div class=\\"column\\"><div class=\\"content has-text-centered\\"><span class=\\"icon has-text-primary\\"><i class=\\"fas fa-tachometer-alt fa-5x\\" aria-hidden=\\"true\\"></i></span><h5 class=\\"title\\">Run your benchmarks</h5></div><div class=\\"content\\"><p>Run your benchmarks locally or in CI using your favorite benchmarking tools. The <code>bencher</code> CLI simply wraps your existing benchmark harness and stores its results.</p><br></div></div></div></div><br><div class=\\"column\\"><div class=\\"columns\\"><div class=\\"column\\"><div class=\\"content has-text-centered\\"><span class=\\"icon has-text-primary\\"><i class=\\"fas fa-chart-line fa-5x\\" aria-hidden=\\"true\\"></i></span><h5 class=\\"title\\">Track your benchmarks</h5></div><div class=\\"content\\"><p>Track the results of your benchmarks over time. Monitor, query, and graph the results using the Bencher web console based on the source branch and testbed.</p></div><br></div></div></div><br><div class=\\"column\\"><div class=\\"columns\\"><div class=\\"column\\"><div class=\\"content has-text-centered\\"><span class=\\"icon has-text-primary\\"><i class=\\"fas fa-bell fa-5x\\" aria-hidden=\\"true\\"></i></span><h5 class=\\"title\\">Catch performance regressions</h5></div><div class=\\"content\\"><p>Catch performance regressions in CI. Bencher uses state of the art, customizable analytics to detect performance regressions before they make it to production.</p></div></div></div></div></div></div></section><hr><br><div class=\\"box\\"><div class=\\"content has-text-centered\\"><h2>Use Your Favorite Tools</h2><br><div class=\\"columns is-centered is-vcentered is-multiline\\"><div class=\\"column is-1\\"><span class=\\"icon has-text-primary is-large\\"><i aria-hidden=\\"true\\" class=\\"devicon-cplusplus-line fa-5x\\"></i></span></div><div class=\\"column is-1\\"><span class=\\"icon has-text-primary is-large\\"><i aria-hidden=\\"true\\" class=\\"devicon-python-plain fa-5x\\"></i></span></div><div class=\\"column is-1\\"><span class=\\"icon has-text-primary is-large\\"><i aria-hidden=\\"true\\" class=\\"devicon-java-plain fa-5x\\"></i></span></div><div class=\\"column is-1\\"><span class=\\"icon has-text-primary is-large\\"><i aria-hidden=\\"true\\" class=\\"devicon-javascript-plain fa-5x\\"></i></span></div><div class=\\"column is-1\\"><span class=\\"icon has-text-primary is-large\\"><i aria-hidden=\\"true\\" class=\\"devicon-ruby-plain fa-5x\\"></i></span></div><div class=\\"column is-1\\"><span class=\\"icon has-text-primary is-large\\"><i aria-hidden=\\"true\\" class=\\"devicon-csharp-line fa-5x\\"></i></span></div><div class=\\"column is-1\\"><span class=\\"icon has-text-primary is-large\\"><i aria-hidden=\\"true\\" class=\\"devicon-go-original-wordmark fa-5x\\"></i></span></div><div class=\\"column is-1\\"><span class=\\"icon has-text-primary is-large\\"><i aria-hidden=\\"true\\" class=\\"devicon-rust-plain fa-5x\\"></i></span></div></div></div></div><br><hr><section class=\\"section\\"><div class=\\"container\\"><div class=\\"columns is-centered\\"><div class=\\"column\\"><h2 class=\\"title\\">Hosting</h2></div></div><br><br><div class=\\"columns is-centered\\"><div class=\\"column\\"><div class=\\"columns\\"><div class=\\"column\\"><div class=\\"content has-text-centered\\"><h2 class=\\"subtitle\\">Self-Hosted</h2><br><br><br><span class=\\"icon has-text-primary\\"><i class=\\"fas fa-server fa-10x\\" aria-hidden=\\"true\\"></i></span></div><div class=\\"content\\"><p>Run Bencher on-prem or in your own cloud. Bencher can be deployed on a standalone server, in a Docker container, or as part of a Kubernetes cluster.</p><br></div><button class=\\"button is-fullwidth\\">Learn More</button></div></div></div><div class=\\"is-divider-vertical\\" data-content=\\"OR\\"></div><div class=\\"column\\"><div class=\\"columns\\"><div class=\\"column\\"><div class=\\"content has-text-centered\\"><h2 class=\\"subtitle\\">Bencher Cloud</h2><br><br><br><span class=\\"icon has-text-primary\\"><i class=\\"fas fa-cloud fa-10x\\" aria-hidden=\\"true\\"></i></span></div><div class=\\"content\\"><p>It's 2023<!---->, who wants to manage yet another serviceâ€½ Let us take care of that for you. All of the same great features with none of the hassle.</p><br></div><button class=\\"button is-fullwidth\\">Get Started</button></div></div></div></div></div></section></section>"`;
